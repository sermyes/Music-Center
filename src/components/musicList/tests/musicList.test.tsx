import MusicList from '../musicList';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { fakeVideos } from '../../tests/videos';
import userEvent from '@testing-library/user-event';

describe('Music List', () => {
  const videos = [...fakeVideos, ...fakeVideos];
  const onVideoClick = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <MusicList videos={videos} onVideoClick={onVideoClick} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('calls onVideoClick with video when clicked', async () => {
    render(<MusicList videos={videos} onVideoClick={onVideoClick} />);
    const videoItem = screen.getAllByRole('img');
    await userEvent.click(videoItem[1]);

    expect(onVideoClick).toHaveBeenCalledTimes(1);
    expect(onVideoClick).toHaveBeenCalledWith(videos[1]);
  });
});
