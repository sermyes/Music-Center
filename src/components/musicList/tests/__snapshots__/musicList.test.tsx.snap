// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Music List renders correctly 1`] = `
Array [
  <div
    className="titleContainer"
  >
    <h2
      className="title"
    >
      Music List
    </h2>
  </div>,
  <div
    className="sliderContainer"
  >
    <div
      className="slider"
    >
      <div
        className="slider simple__slider"
      >
        <div
          className="slidesContainer"
          data-testid="container"
        >
          <div
            className="slides"
            data-testid="slides"
          >
            <div
              className="slide"
            >
              <div
                className="container"
              >
                <div
                  className="video"
                  onClick={[Function]}
                  title="video"
                >
                  <img
                    alt="title1"
                    className="thumbnail"
                    src="https://image/"
                  />
                  <div
                    className="metadata"
                  >
                    <p
                      className="title"
                    >
                      title1
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="slide"
            >
              <div
                className="container"
              >
                <div
                  className="video"
                  onClick={[Function]}
                  title="video"
                >
                  <img
                    alt="title2"
                    className="thumbnail"
                    src="https://image/"
                  />
                  <div
                    className="metadata"
                  >
                    <p
                      className="title"
                    >
                      title2
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="slide"
            >
              <div
                className="container"
              >
                <div
                  className="video"
                  onClick={[Function]}
                  title="video"
                >
                  <img
                    alt="title3"
                    className="thumbnail"
                    src="https://image/"
                  />
                  <div
                    className="metadata"
                  >
                    <p
                      className="title"
                    >
                      title3
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="slide"
            >
              <div
                className="container"
              >
                <div
                  className="video"
                  onClick={[Function]}
                  title="video"
                >
                  <img
                    alt="title1"
                    className="thumbnail"
                    src="https://image/"
                  />
                  <div
                    className="metadata"
                  >
                    <p
                      className="title"
                    >
                      title1
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="slide"
            >
              <div
                className="container"
              >
                <div
                  className="video"
                  onClick={[Function]}
                  title="video"
                >
                  <img
                    alt="title2"
                    className="thumbnail"
                    src="https://image/"
                  />
                  <div
                    className="metadata"
                  >
                    <p
                      className="title"
                    >
                      title2
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="slide"
            >
              <div
                className="container"
              >
                <div
                  className="video"
                  onClick={[Function]}
                  title="video"
                >
                  <img
                    alt="title3"
                    className="thumbnail"
                    src="https://image/"
                  />
                  <div
                    className="metadata"
                  >
                    <p
                      className="title"
                    >
                      title3
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="arrow"
        >
          <button
            className="prev"
            data-testid="arrow"
            onClick={[Function]}
          >
            <i
              className="icon fas fa-chevron-left"
            />
          </button>
          <button
            className="next"
            data-testid="arrow"
            onClick={[Function]}
          >
            <i
              className="icon fas fa-chevron-right"
            />
          </button>
        </div>
        <ul
          className="dots simple__dots"
          onClick={[Function]}
        />
      </div>
    </div>
  </div>,
]
`;
